docs/proposals/device-crd.md

```diff
@@ -0,0 +1,556 @@
+  // Required.
+  Metadata     map[string]string `json:"metadata,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DeviceModel is the Schema for the device model API
+// +k8s:openapi-gen=true
+type DeviceModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec DeviceModelSpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DeviceModelList contains a list of DeviceModel
+type DeviceModelList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DeviceModel `json:"items"`
+}
+```
+
+### Device model sample
+```yaml
+apiVersion: devices.kubeedge.io/v1alpha1
+kind: DeviceModel
+metadata:
+  labels:
+    description: 'TI Simplelink SensorTag Device Attributes Model'
+    manufacturer: 'Texas Instruments'
+    model: CC2650
+  name: sensor-tag-model
+spec:
+  properties:
+  - name: temperature
+    description: 'Temperature from Temperature Sensor in Degrees Fahrenheit'
+    metadata:
+      dataType: float
+      accessMode: r
+      unit: 'degree fahrenheit'
+  - name: temperature-enable
+    description: 'Enable Temperature Sensor'
+    metadata:
+      dataType: string
+      accessMode: rw
+  propertyVisitors:
+    - protocol:  BluetoothLE # valid values : [BluetoothLE, Zigbee, ModbusTCP, ModbusRTU, BACnet, OPCUA]
+      visitors:
+      - name: temperature
+        propertyName: temperature
+        visitorConfig:
+          uuid: F000AA01-0451-4000-B000-000000000000
+      - name: temperatureEnable
+        propertyName: temperature-enable
+        visitorConfig:
+          uuid: F000AA02-0451-4000-B000-000000000000
+```
+
+The above example is for a [CC2650 SensorTag](http://processors.wiki.ti.com/index.php/CC2650_SensorTag_User's_Guide) device. The SensorTag exposes all its sensors via [Bluetooth Smart (Bluetooth Low Energy)](https://en.wikipedia.org/wiki/Bluetooth_Low_Energy) services. The sensor drivers interface to a GATT server running on TI BLE-Stack v2. The GATT server contains a primary service for each sensor for configuration and data collection. Each service exposes charactersitics which can be accessed via 128 bit UUIDs.
+
+The sample shows two charactersitics being exposed to the user. Access modes describe whether the charactersitic is read-only or can be written to. For the temperature sensor, the temperature is a read-only charactersitic . The sensor can be configured to enable data collection via another charactersitic.
+
+Property visitors provide such additional details like how to access these device properties. For example, the temperature can be read from charactersitic with UUID `F000AA01-0451-4000-B000-000000000000`. If we need to enable data collection, we can write to the charactersitic with UUID `F000AA02-0451-4000-B000-000000000000`.
```

Comment:
Visitor seems very tricky to me and the meaningful field of Visitor are only propertyName and a map structure. Can we reduce the concepts?

Reply:
@zhenghaibo1008 , please provide your thoughts ?

Reply:
both attribute and property are fine.In c#,  attribute is very simple, just with the type definition. But property is complicate， it has getter and setter. In our scenario, maybe property is more suitable

Reply:
OK, I buy it :)